### ISSUE TITLE
False positive: sqlRowsClose: local variable db.Rows have not Close call

### ISSUE THREAD

**Issue by AnikHasibul** [go-critic/go-critic#609]
```go
rows, err := db.Query("...", foo)
if err != nil {
       return 0, err
}
defer func() {
       // closing here
        err := rows.Close()
        if err != nil {
                panic(err)
        }                                         
}()                                                

for rows.Next() {
      //....
}
```

In this code I've closed `rows` in a defer call. 
But it's still reporting errors.

```txt
sqlRowsClose: local variable db.Rows have not Close call
```

**Comment by Quasilyte** [comment]
Thanks for reporting it!
Will be fixed as a part of the current milestone.

**Comment by AnikHasibul** [comment]
@Quasilyte @cristaloleg  It's still a bug!
And still giving the same error. Please scan the snippet given on the issue.
And it's really an annoying error! Because it reports something related to the database!
And the real annoying thing is the error message is a false positive!
Please have a look at it!

**Comment by cristaloleg** [comment]
We do not scan assign state properly, simple reproducer:
```
type authUser struct {
	db *sql.DB
}

func (a *authUser) auth(foo string) (uid int64, err error) {
	rows, err := a.db.Query("...", foo)
	if err != nil {
		return 0, err
	}
	defer func() {
		err := rows.Close()
		if err != nil {
			panic(err)
		}
	}()
	if err := rows.Close(); err != nil {
		return 0, err
	}
	return 42, nil
}

```
